modules/codegen-plugin/src/smithy4s/codegen/CodegenPlugin.scala

```diff
@@ -172,6 +176,16 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
         .toList
 
       resPaths.map(path => new File(path.toString))
-    } else (conf / smithy4sOutputDir).value.listFiles().toSeq
+    } else {
+      val base = (conf / smithy4sOutputDir).value
+
+      val allOutputFiles = (base ** "*").get()
```

Comment:
I'm not certain that the logic is correct, considering smithy4s doesn't wipe the output folder between runs. Can't we use a similar mechanism to the one you used, but for outputs ?